services/provider/server/server.go

```diff
@@ -577,7 +577,292 @@ func (s *OCSProviderServer) RevokeStorageClaim(ctx context.Context, req *pb.Revo
+		}
+		consumerConfigMap.Name = storageConsumer.Status.ResourceNameMappingConfigMap.Name
+		consumerConfigMap.Namespace = storageConsumer.Namespace
+		err = s.client.Get(ctx, client.ObjectKeyFromObject(consumerConfigMap), consumerConfigMap)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get %s configMap. %v", consumerConfigMap.Name, err)
+		}
+		kubeResources, err := s.getKubeResources(ctx, storageConsumer)
+		if err != nil {
+			return nil, err
+		}
+
+		consumerConfig := util.WrapStorageConsumerResourceMap(consumerConfigMap.Data)
+
+		storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		if req.StorageClaimName == fmt.Sprintf("%s-ceph-rbd", storageCluster.Name) {
+			extR, err = s.getStorageClaimConfigBlock(req, kubeResources,
+				storageCluster.Name,
+				consumerConfig.GetCsiRbdProvisionerCephUserName(),
+				consumerConfig.GetCsiRbdNodeCephUserName(),
+				consumerConfig.GetRbdClientProfileName(),
+			)
+			if err != nil {
+				klog.Errorf("failed to get storage claim config. %v", err)
+				return nil, status.Errorf(codes.Internal, "failed to get storage claim config. %v", err)
+			}
+		} else if req.StorageClaimName == fmt.Sprintf("%s-cephfs", storageCluster.Name) {
+			extR, err = s.getStorageClaimConfigCephFs(req, kubeResources,
+				storageCluster.Name,
+				consumerConfig.GetCsiCephFsProvisionerCephUserName(),
+				consumerConfig.GetCsiRbdNodeCephUserName(),
+				consumerConfig.GetCephFsClientProfileName(),
+			)
+			if err != nil {
+				klog.Errorf("failed to get storage claim config. %v", err)
+				return nil, status.Errorf(codes.Internal, "failed to get storage claim config. %v", err)
+			}
+		}
+	}
+
+	klog.Infof("successfully returned the storage class claim %q for %q", req.StorageClaimName, req.StorageConsumerUUID)
+	return &pb.StorageClaimConfigResponse{ExternalResource: extR}, nil
+
+}
+
+// SHIM for GetStorageClaimConfig to return response to 4.18 clients
+func (s *OCSProviderServer) getStorageClaimConfigBlock(
+	req *pb.StorageClaimConfigRequest,
+	kubeResources []client.Object,
+	storageClusterName string,
+	provSecretName string,
+	nodeSecretName string,
+	clientProfileName string,
+) ([]*pb.ExternalResource, error) {
+
+	var extR []*pb.ExternalResource
+
+	for _, resource := range kubeResources {
+		gvk, err := apiutil.GVKForObject(resource, s.scheme)
+		if err != nil {
+			return nil, err
+		}
+
+		switch gvk {
+		case v1.SchemeGroupVersion.WithKind("Secret"):
+			secret := resource.(*corev1.Secret)
+			provSecName := provSecretName
+			nodeSecName := nodeSecretName
+			if secret.Name == provSecName || secret.Name == nodeSecName {
+				extR = append(extR, &pb.ExternalResource{
+					Name:        secret.Name,
+					Kind:        gvk.Kind,
+					Labels:      resource.GetLabels(),
+					Annotations: resource.GetAnnotations(),
+					Data:        mustMarshal(secret.Data),
+				})
```

Comment:
Old secret format is missing here

Reply:
Why is the old secret format required here?
https://github.com/red-hat-storage/ocs-operator/pull/3145#discussion_r2049446914